src/btree/bt_sync.c

```diff
@@ -535,7 +606,9 @@ __wt_sync_file(WT_SESSION_IMPL *session, WT_CACHE_OP syncop)
                 break;
 
             if (F_ISSET(walk, WT_REF_FLAG_INTERNAL) && internal_cleanup) {
-                WT_WITH_PAGE_INDEX(session, ret = __sync_ref_int_obsolete_cleanup(session, walk));
+                __sync_load_throttle_init(&load_throttle);
+                WT_WITH_PAGE_INDEX(
+                  session, ret = __sync_ref_int_obsolete_cleanup(session, walk, &load_throttle));
```

Comment:
The `__sync_load_throttle_init` call could be moved into `__sync_ref_int_obsolete_cleanup()`, and combined with the `__sync_load_throttle_next_int` call as far as I can tell, and there's no reason we couldn't just do the initialization inline there?

This seems more complicated that it needs to be: `__sync_should_read_obsolete_page()` can be combined with `__sync_load_throttle_check()`, too.

IOW, I think we only need a single helper function.

Reply:
The goal was to have the entire throttle computation in one place, because it's currently arbitrary stuff I made up and so it's reasonably likely that someone sometime will want to rearrange it. Also, it turns out the init call was in the wrong place; it was supposed to be before the treewalk starts, not once per page, so those two pieces aren't actually the same. (Fixed this in 7b54c135e.) (Whole thing obsoleted in 395664db3.)